pkg/metrics/providers/graphite_test.go

```diff
@@ -124,6 +172,92 @@ func TestNewGraphiteProvider(t *testing.T) {
 	}
 }
 
+func TestGraphiteProvider_RunQuery(t *testing.T) {
+	tests := []struct {
+		name           string
+		query          string
+		expectedTarget string
+		expectedResult float64
+		errExpected    bool
+		body           string
+	}{{
+		"ok",
+		"target=sumSeries(app.http.*.*.count)&from=-2min",
+		"sumSeries(app.http.*.*.count)",
+		float64(100),
+		false,
+		`[
+			{
+				"datapoints": [
+					[
+						10,
+						1621348400
+					],
+					[
+						75,
+						1621348410
+					],
+					[
+						25,
+						1621348420
+					],
+					[
+						100,
+						1621348430
+					]
+				],
+				"target": "sumSeries(app.http.*.*.count)",
+				"tags": {
+					"aggregatedBy": "sum",
+					"name": "sumSeries(app.http.*.*.count)"
+				}
+			}
+		]`,
```

Comment:
This looks good. Nonblocking though: should we also be testing scenarios exercising `UnmarshalJSON`'s error branches? Or what if the query passed to `RunQuery` has a `?format=json` and/or other query params?

Reply:
We certainly can add more tests, but TBH, I think many of the code paths are "just in case"... I'm not sure the Graphite render API can/will actually generate payloads that would exercise those paths. Another option to raise coverage may be to cut out or simplify some of those code paths.

I don't think we need to re-test the standard library's [`Values`](https://golang.org/pkg/net/url/#Values) API.